tests/common/widgets/rating-menu/options.ts

```diff
@@ -0,0 +1,447 @@
+                    class="ais-RatingMenu-starIcon ais-RatingMenu-starIcon--empty"
+                    height="24"
+                    width="24"
+                  >
+                    <use
+                      href="#ais-RatingMenu-starEmptySymbol"
+                    />
+                  </svg>
+                  <svg
+                    aria-hidden="true"
+                    class="ais-RatingMenu-starIcon ais-RatingMenu-starIcon--empty"
+                    height="24"
+                    width="24"
+                  >
+                    <use
+                      href="#ais-RatingMenu-starEmptySymbol"
+                    />
+                  </svg>
+                  <svg
+                    aria-hidden="true"
+                    class="ais-RatingMenu-starIcon ais-RatingMenu-starIcon--empty"
+                    height="24"
+                    width="24"
+                  >
+                    <use
+                      href="#ais-RatingMenu-starEmptySymbol"
+                    />
+                  </svg>
+                  <svg
+                    aria-hidden="true"
+                    class="ais-RatingMenu-starIcon ais-RatingMenu-starIcon--empty"
+                    height="24"
+                    width="24"
+                  >
+                    <use
+                      href="#ais-RatingMenu-starEmptySymbol"
+                    />
+                  </svg>
+                  <span
+                    aria-hidden="true"
+                    class="ais-RatingMenu-label"
+                  >
+                    & Up
+                  </span>
+                  <span
+                    class="ais-RatingMenu-count"
+                  >
+                    18047
+                  </span>
+                </a>
+              </div>
+            </li>
+          </ul>
+        </div>
+      `
+      );
+    });
+
+    test('limits the number of stars to display', async () => {
+      const searchClient = createMockedSearchClient();
+
+      await setup({
+        instantSearchOptions: {
+          searchClient,
+          indexName: 'indexName',
+        },
+        widgetParams: {
+          attribute: 'rating',
+          max: 3,
+        },
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      const items = document.querySelectorAll('.ais-RatingMenu-item');
+
+      // `max` is exclusive, the number of items is max - 1
+      expect(items).toHaveLength(2);
+
+      items.forEach((item) =>
+        expect(item.querySelectorAll('.ais-RatingMenu-starIcon')).toHaveLength(
+          3
+        )
+      );
+    });
+  });
+}
+
+function createMockedSearchClient() {
+  return createSearchClient({
+    search: jest.fn((requests) => {
+      return Promise.resolve(
+        createMultiSearchResponse(
+          ...requests.map(() =>
+            createSingleSearchResponse({
+              facets: {
+                rating: {
+                  0: 3422,
```

Comment:
isn't there float logic in the rating menu too? do you want to test those cases here or in other tests?

Reply:
I see some logic in the connector but no tests all around. The doc also mentions that attribute values should be integers. Is it still relevant?

Reply:
I looked more but still can't find the use case for floats in rating menu: we don't have fragments of stars to render 2.5/5 stars for example, and the count reflects the number of items in the dataset so it can only be an integer.

Is there something else I'm missing?

Reply:
I think it's rounded, or maybe that was in numeric menu or range.